### PR TITLE
[metricsadvisor] Update async samples to use asyncio.run()

### PR DESCRIPTION
**Pull request by sammiee5311**
# Description

related to #21211

drop `loop.run_untill_complete` to use `asyncio.run()`.

Thanks!

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by kristapratico — APPROVED** [review]
Hey @sammiee5311, thanks for your PR! This looks good to me, but I'm going to wait for @xiangyan99's review before merging. :)

**Review by xiangyan99 — APPROVED** [review]
The PR looks good to me. Thanks for the contribution!

**Pull request merged**
